### ISSUE TITLE
Support Azure Spot VM

### ISSUE THREAD

**Issue by yzgyyang** [cirruslabs/cirrus-ci-docs#778]
# Description

Similar to the GCP products used, Azure Spot VM is the preemptible type of instances suitable for CI - I would personally like to see this support so that I can spin up FreeBSD VMs with BYO Infrastructure.

# Context

# Anything Else

...

**Comment by fkorotkov** [comment]
Does Azure have FreeBSD!? 😮

**Comment by yzgyyang** [comment]
> Does Azure have FreeBSD!? 😮

Yes! They have up-to-date FreeBSD images by Microsoft. I've been using it for years already! And now I'm considering moving my build jobs to Azure as well, hence this feature request.

https://docs.microsoft.com/en-us/azure/virtual-machines/linux/freebsd-intro-on-azure
